src/main/java/mekanism/common/CommonEventHandler.java

```diff
@@ -0,0 +1,32 @@
+package mekanism.common;
+
+import mekanism.common.item.ItemJetpack;
+import net.minecraftforge.fml.common.Mod.EventHandler;
+import net.minecraftforge.fml.common.gameevent.TickEvent;
+import net.minecraftforge.fml.relauncher.Side;
+
+/**
+ * @author ThatGamerBlue
+ */
+public class CommonEventHandler {
+
+    private boolean canFly = false;
+
+    @EventHandler
+    public void onPlayerTick(TickEvent.PlayerTickEvent event){
+        if(event.phase == TickEvent.Phase.START && event.side == Side.SERVER){
+            if(canFly){
+                if(!(event.player.inventory.armorInventory[1].getItem() instanceof ItemJetpack)){
+                    canFly = false;
+                    event.player.capabilities.allowFlying = canFly;
```

Comment:
Basing it solely on mek jetpack would probably break any other mod changing this wouldn't it?